### DISCUSSION TITLE
Change ID format

### DISCUSSION THREAD

**Discussion by redwiz666** [cmu-sei/GHOSTS#426] [Q&A] [answered]
Good morning. I am trying to run ghosts within 5 subnets that are all part of a larger range. The issue I'm running into is that all the subnets are mostly identical to each other except for the IP. Ghost seems to be thinking these are all the same machines and the Ids being generated are identical. Looking into the application.conf  I see the ID format be specified. Is there a list of options that we can use to modify this format?

**Comment by sei-dupdyke** [comment, marked as answer]
Hi! Do you happen to be running these on vmware? If so, you can use [guestinfovars](https://github.com/cmu-sei/GHOSTS/blob/d7977c02c8ea0808bb43f4d4dba9ad9afad979b2/src/Ghosts.Client/Infrastructure/GuestInfoVars.cs#L10) - the "format key" or name of the guestinfovar is set in the app configuration file ./config/application.json on [this line](https://github.com/cmu-sei/GHOSTS/blob/d7977c02c8ea0808bb43f4d4dba9ad9afad979b2/src/Ghosts.Client/config/application.json#L10). The formatvalue right below is where that value gets set in the machine name. 

So if we set these to:

"FormatKey": "guestinfo.rangename",
"FormatValue": "$formatkeyvalue$-$machinename$",

and in vmware we create a guestinforvar named "guestinfo.rangename" and set it to our range name "R_1", and the machine is M_1, the name will be "R_1-M_1" in ghosts.

If not on vmware, we'll have to think of a quick fix up. Please lmk.

**Reply by redwiz666** [reply]
That's for the reply, yea I'm using VMware. So your saying I can change these variables to a hard coded value such as range_1 and it will use it in the Id field? 
 I tried changing the id.json in the instance folder but that didn't have any effect

**Reply by redwiz666** [reply]
Ok i think i figured it out. I ran c:\program files\VMware\VMware Tools\vmtoolsd.exe --cmd "info-set guestinfo.rangename range1" and now its pulling a unique id for the host. 

Thank you for the reply!
